ingest: document fields that support templating

jakelandis opened the pull request:
This change also updates many of the examples to use ecs as the example.
Some additional minor improvements are also included.

Part of #33188

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -751,23 +751,23 @@ pipeline basis. Useful to find out which pipelines are used the most or spent th
 Appends one or more values to an existing array if the field already exists and it is an array.
 Converts a scalar to an array and appends one or more values to it if the field exists and it is a scalar.
 Creates an array containing the provided values if the field doesn't exist.
```
<img width="767" alt="image" src="https://user-images.githubusercontent.com/976291/47042762-b724d280-d151-11e8-95da-a02a4b0320be.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -846,7 +846,8 @@ Parses dates from fields, and then uses the date or timestamp as the timestamp f
 By default, the date processor adds the parsed date as a new field called `@timestamp`. You can specify a
```
<img width="774" alt="image" src="https://user-images.githubusercontent.com/976291/47042851-e1769000-d151-11e8-93ce-fe2b63799d45.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -1041,12 +1042,12 @@ understands this to mean `2016-04-01` as is explained in the <<date-math-index-n
 |======
```
<img width="771" alt="image" src="https://user-images.githubusercontent.com/976291/47042953-184ca600-d152-11e8-9623-687d08c52534.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -1365,21 +1366,22 @@ a scalar field to an object field.
 === Fail Processor
 Raises an exception. This is useful for when
```
<img width="768" alt="image" src="https://user-images.githubusercontent.com/976291/47043003-2c90a300-d152-11e8-8ccf-b2cf2cc4f13e.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -2077,14 +2079,14 @@ Converts a string to its lowercase equivalent.
 
 [[remove-processor]]
```
<img width="788" alt="image" src="https://user-images.githubusercontent.com/976291/47043039-4205cd00-d152-11e8-978e-b32f50780151.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -2115,23 +2117,24 @@ To remove multiple fields, you can use the following query:
 
 [[rename-options]]
 .Rename Options
 [options="header"]
```
<img width="788" alt="image" src="https://user-images.githubusercontent.com/976291/47043083-55189d00-d152-11e8-846d-01f198ef82b0.png">

jakelandis commented on docs/reference/ingest/ingest-node.asciidoc:
```diff
@@ -2243,24 +2246,24 @@ and depending on the progress made, indexed into different indices.
 [[set-processor]]
 === Set Processor
```
<img width="771" alt="image" src="https://user-images.githubusercontent.com/976291/47043129-75e0f280-d152-11e8-8583-a07932a7167d.png">

jakelandis replied:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/976291/47232023-2ee23f80-d394-11e8-84c7-946c705ebba9.png">

@tomcallahan @debadair ^^ example using description instead of superscript. Thoughts on which you prefer seeing side by side ?

tomcallahan replied:
I like this the best of the available options

elasticmachine commented:
Pinging @elastic/es-core-infra

jakelandis commented:
@debadair - could I get your review on this documentation change ?  I am abit uncertain about the superscript approach, however after playing around with a few options it came out as less busy/crowded as other options I tried. I have included screen shots of the rendered output to hopefully make the review quick. 

Also, I updated many of the examples to be based from https://github.com/elastic/ecs where applicable.

tomcallahan commented:
So you're saying it didn't look good with "templating supported" as another column in the table?  I am not a huge fan the superscript

jakelandis commented:
> So you're saying it didn't look good with "templating supported" as another column in the table? I am not a huge fan the superscript.

It's a judgement call for sure ... here is an example of the set processor rendered with the extra column:
<img width="1190" alt="image" src="https://user-images.githubusercontent.com/976291/47164017-c5433200-d2bc-11e8-8b17-1044061d2532.png">

> It also appears that field and target_field are templatable in all cases exception one -- can you verify that is correct?

Of the processors changed, Date and Date Index Name do not support templates for `field`or `target_field`. There are a couple processors _not_ part of the changes here that also do not support templates for `field` or `target_field`

jakelandis commented:
@debadair @tomcallahan - Thanks for the feedback!  I have removed the superscript in favor of a comment in the description.  (and updated the screen shots)

jakelandis commented:
@tomcallahan - An oversight on my part, I will update the wording to be consistent.

jakelandis commented:
jenkins test this please

jakelandis commented:
6.5 backport: https://github.com/elastic/elasticsearch/commit/e9a380c76fae13ebaee7401765e27a273dfd22d0
